src/js/letters/actions/letters.js

```diff
@@ -12,9 +12,25 @@ import {
   GET_LETTER_PDF_FAILURE,
   GET_LETTER_PDF_SUCCESS,
   LETTER_ELIGIBILITY_ERROR,
-  UPDATE_BENFIT_SUMMARY_REQUEST_OPTION
+  UPDATE_BENFIT_SUMMARY_REQUEST_OPTION,
+  UPDATE_ADDRESS_SUCCESS,
+  UPDATE_ADDRESS_FAILURE
 } from '../utils/constants';
 
+// Copied from the vets-api v0/address POST swagger; TODO: delete once
+// actual form data is passed to updateAddress
+const exampleAddress = {
```

Comment:
@kreek This address is lifted from the vets-api address swagger but fails validation (returns a 422)

Reply:
@raquelromano-gov I think this is an olive branch issue, without the header `"X-Key-Inflection: camel"` that converts camel to snake case I could reproduce the 422, if I set the header then it goes through.

```
curl -X "PUT" "http://localhost:3000/v0/address" \
     -H "X-Key-Inflection: camel" \
     -H "Authorization: Token token=ZQfL15JDekMLBJyvRutJKzS8dCeWYwEjyrq9XhxK" \
     -H "Content-Type: application/json; charset=utf-8" \
     -d $'{
  "zipCode": "97477",
  "city": "Springfield",
  "addressTwo": "",
  "addressEffectiveDate": "2017-08-07T19:43:59.383Z",
  "zipSuffix": "",
  "countryName": "USA",
  "type": "DOMESTIC",
  "addressOne": "225 5th St",
  "addressThree": "",
  "stateCode": "OR"
}'
```
Tested this on DEV (vs INT) and could update an address. Seems like you can set this [per request](https://github.com/department-of-veterans-affairs/vets-website/blob/5e51527b04d4e3fb0eb66c11b2eff077c2c37202/src/js/edu-benefits/1990/actions/index.js#L159) or in [a config](https://github.com/department-of-veterans-affairs/vets-website/blob/5e51527b04d4e3fb0eb66c11b2eff077c2c37202/src/js/edu-benefits/1990/actions/index.js#L159).

Reply:
@kreek Thanks for uncovering this! Looks like it's actually a default header for our `GET` requests but I was overriding those for a `POST`, `PUT`, or `PATCH`, so I will update our common JS to default to this as well: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/common/helpers/api.js#L20

Reply:
@kreek Could use your eyeballs again. Now I'm getting a 500

Relevant  vets-api server log:
```
/Users/romano/dsva/vets-api/lib/evss/pciu_address/service.rb:39:in `block in update_address'
/Users/romano/dsva/vets-api/lib/evss/pciu_address/service.rb:47:in `with_exception_handling'
/Users/romano/dsva/vets-api/lib/evss/pciu_address/service.rb:31:in `update_address'
/Users/romano/dsva/vets-api/app/controllers/v0/addresses_controller.rb:13:in `update'
```
https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/evss/pciu_address/service.rb#L38

Inspecting `address` here with byebug, I see:
```
{"type"=>"DOMESTIC", "address_effective_date"=>"2017-08-29T05:12:51.300+00:00", "address_one"=>"225 5th St", "city"=>"Springfield", "country_name"=>"USA", "state_code"=>"OR", "zip_code"=>"97477"}
```

Request body from Chrome dev tools:
```
{"zipCode":"97477","city":"Springfield","zipSuffix":"","countryName":"USA","type":"DOMESTIC","addressOne":"225 5th St","stateCode":"OR"}
```

Request headers from dev tools:
```
PUT /v0/address HTTP/1.1
Host: localhost:3000
Connection: keep-alive
Content-Length: 136
Pragma: no-cache
Cache-Control: no-cache
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.101 Safari/537.36
Origin: http://localhost:3001
X-Key-Inflection: camel
Authorization: Token token=seddcHi8K3MiyDuZA3aRohE4n6jRXkRHLJ9dmSw_
Content-Type: application/json; charset=utf-8
Accept: */*
Referer: http://localhost:3001/letters/confirm-address
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8
```

Can you LMK if anything jumps out as strange to you?